### PR TITLE
[github] Update the pull request template

### PR DESCRIPTION
**Pull request by sgnn7**
### What does this PR do?

Updates the PR template for GitHub:
- Old template suggestions were moved to invisible comments
- Additional information was added about how to fill each section
- Additional sections were added as needed
- Checklist has ben updated with:
  - The new items and the order of precedence
  - Wording to indicate that its use is primarily for the reviewer

### Motivation

The current PR template has been inadequate as the template
was usually not filled out, leading to problems QAing/verifying
patches along with not having clearer guidelines for the reviewer. 

When no information has been filled in, the new template makes it obvious when content is missing:

<hr />
 
 Screen Shot 2021-11-05 at 2 13 58 PM

<hr />

### Additional Notes

N/A

### Describe how to test your changes

N/A

### Checklist

- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,31 +1,67 @@
...
-Anything else we should know when reviewing?
+<!--
+* Anything else we should know when reviewing?
+* Include benchmarking information here whenever possible.
+* Include info about alternatives that were considered and why the proposed
+  version was chosen.
+-->
 
-### Describe how to test your changes
+### Possible Drawbacks
```
Comment (truthbk) [inline .github/PULL_REQUEST_TEMPLATE.md:36]:
```suggestion
### Possible Drawbacks / Trade-offs
```

**Review by truthbk — APPROVED** [review]
I'm good with this, added a suggestion, but this already LGTM! ✨

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (sgnn7) [reply .github/PULL_REQUEST_TEMPLATE.md]: 👍

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,31 +1,67 @@
...
+* If the description here cannot be expressed in a succint form, consider
+  opening multiple pull requests instead of a single one.
+-->
 
 ### Motivation
 
-What inspired you to submit this pull request?
+<!--
+* What inspired you to submit this pull request?
+* Link any related GitHub issues here.
```
Comment (albertvaka) [inline .github/PULL_REQUEST_TEMPLATE.md:24]:
```suggestion
* Link any related GitHub issues or PRs here.
```
I think sometimes we miss links between PRs (eg: follow-up PRs that fix edge cases from previous PRs).

**Review by albertvaka — APPROVED** [review]

**Review by gh123man — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,31 +1,67 @@
...
+
+- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
+- [ ] The appropriate `team/..` label has been applied, if known.
+- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno)
+  has been added or the `changelog/no-changelog` label has been applied.
+- [ ] Changed code has automated tests for its functionality.
+- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
+- [ ] If applicable, docs team has been notified or
+  [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
+- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels has been applied.
```
Comment (djmitche) [inline .github/PULL_REQUEST_TEMPLATE.md:65]:
```suggestion
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
```

**Review by djmitche — APPROVED** [review]

**Pull request merged**